### PR TITLE
{ai}[gfbf/2023a] Safetensors v0.4.3

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

needs easyblock PR:
- https://github.com/easybuilders/easybuild-easyblocks/pull/3341

### CODE REVIEW

**Comment by smoors** [comment]
Test report by @smoors
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3341
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node403.hydra.os - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7282 16-Core Processor (zen2), 1 x NVIDIA NVIDIA A100-PCIE-40GB, 550.54.15, Python 3.6.8
See https://gist.github.com/smoors/5dfeb563cf9ea3e255253b58f859fc3d for a full test report.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso EB_ARGS="--include-easyblocks-from-pr 3341"

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    ('maturin', '1.4.0', '-Rust-' + _rustver),
+]
+
+dependencies = [
+    ('CUDA', '12.1.1', '', SYSTEM),
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('PyTorch', '2.1.2', versionsuffix),  # optional
+    # ('jax', 'xxx', versionsuffix),  # optional, add 'jax' to 'use_pip_extras'
+    # ('TensorFlow', 'xxx', versionsuffix),  # optional, add 'tensorflow' to 'use_pip_extras'
```
Comment (Micket) [inline easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb:28]:
```suggestion
```

I think these can be opt-in at runtime, which would safe us from having to make multiple versions of this (and considering the combinatorics of 3 optional things, with cuda and non-cuda, would be insane, n=2x2x2x2=ugh) 

I skimmed the source code, and it's just different plain python code wrappers for save_file and such in safetensors, and I think they are all installed unconditionally and can be combined with any (compatible) torch/numpy/tensorflow/paddle/etc.etc.

(I could be wrong, didn't test this theory yet)

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+_rustver = '1.75.0'
+
+homepage = 'https://huggingface.co/docs/safetensors'
+description = """Safetensors is a new simple format for storing tensors safely (as opposed to
+pickle) and that is still fast (zero-copy). Safetensors is really fast.
+
+This variant of Safetensors is installed with support for numpy and PyTorch
+"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
```
Comment (Micket) [inline easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb:15]: would also drop this to gfbf when keeping only SciPy-bundle dep, and drop the CUDA and suffix of course

File: easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    {'windows_x86_64_gnullvm-0.52.5.tar.gz': '852298e482cd67c356ddd9570386e2862b5673c85bd5f88df9ab6802b334c596'},
+    {'windows_x86_64_msvc-0.48.5.tar.gz': 'ed94fce61571a4006852b7389a063ab983c02eb1bb37b47f8272ce92d06d9538'},
+    {'windows_x86_64_msvc-0.52.5.tar.gz': 'bec47e5bfd1bff0eeaf6d8b485cc1074891a197ab4225d504cb7a1ab88b02bf0'},
+]
+
+use_pip = True
+
+exts_list = [
+    ('safetensors', version, {
+        'use_pip_extras': 'numpy,torch',
```
Comment (Micket) [inline easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb:294]: In my testing, this also changes nothing in the installation, even just putting gibberish here, I end up with the exact same installation.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/e206349d9f7b8574bde208b660d29447 for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Going in, thanks @smoors!

**Pull request merged**